test/integration/test-user-error-reporting.js

```diff
@@ -83,4 +83,36 @@ describe.configure().run('user-error', function() {
       return expect(withdrawRequest(env.win, randomId)).to.be.ok;
     });
   });
+
+  describes.integration('3p user-error integration test', {
+    extensions: ['amp-analytics', 'amp-ad'],
+    hash: 'log=0',
+    experiments: ['user-error-reporting'],
+
+    body: () => `
+    <amp-ad width=300 height=250
+        type="_ping_"
+        data-url='not-exist'
+        data-valid='false'
+        data-error='true'>
+    </amp-ad>
+
+    <amp-analytics><script type="application/json">
+    {
+      "requests": {
+        "error": "${depositRequestUrl(randomId)}"
+      },
+      "triggers": {
+        "userError": {
+          "on": "user-error",
+          "request": "error"
+        }
+      }
+    }
+    </script></amp-analytics>`,
+  }, env => {
+    it('should ping correct host with 3p error message', () => {
+      return expect(withdrawRequest(env.win, randomId)).to.eventually.be.ok;
```

Comment:
`chai-as-promised` assertions aren't available for integration tests. See test/chai-as-promised/chai-as-promised.js.

Reply:
To clarify, you still can use regular chai assertions. So the ones in http://chaijs.com/api/bdd/ are available in integration tests, but the ones in https://github.com/domenic/chai-as-promised are not available.

Reply:
Thanks @rsimha-amp I fixed the test by return a promise w/o using chai assertions.
I found that `expect(promise).to.be.ok` will always return true. is it true that it equals to  `expect(promise).to.be.defined`? Asking because I see we use `expect(promise).to.be.ok` else where, want to make sure we check the correct things in tests.

Reply:
`to.be.defined` is invalid. See my recent email to the team :)